src/SFA.DAS.LevyTransferMatching.Api/Controllers/FunctionsController.cs

```diff
@@ -104,5 +105,37 @@ public async Task<IActionResult> DebitApplication(DebitApplicationRequest reques
                 return new StatusCodeResult((int)HttpStatusCode.InternalServerError);
             }
         }
+
+        [Route("application-funding-declined")]
+        [HttpPost]
+        public async Task<IActionResult> ApplicationFundingDeclined(ApplicationFundingDeclinedRequest request)
+        {
+            try
+            {
+                var result = await _mediator.Send(new CreditPledgeCommand
+                {
+                    PledgeId = request.PledgeId,
+                    Amount = request.Amount,
+                    ApplicationId = request.ApplicationId
+                });
+
+                if (result == null)
```

Comment:
I'm not keen on returning null as the result of a command, when there is a result object we could use, as this would be unexpected. Couldn't we just add a new property that we can check here to say that the credit was skipped?

Reply:
Done.